src/jetstream_types.rs

```diff
@@ -553,23 +584,215 @@ pub struct NextRequest {
+
+    /// Configures a Consumer to receive
+    /// messages from a start sequence.
+    pub fn deliver_by_start_sequence(mut self, seq: u64) -> Self {
+        self.deliver_policy = Some(DeliverPolicy::ByStartSeq);
+        self.opt_start_seq = Some(seq);
+        self
+    }
+
+    /// Configures the consumer to receive
+    /// messages from a start time.
+    pub fn deliver_by_start_time(mut self, time: DateTime) -> Self {
+        self.deliver_policy = Some(DeliverPolicy::ByStartTime);
+        self.opt_start_time = Some(time);
+
+        self
+    }
+
+    /// Require no acks for delivered messages.
+    pub fn ack_none(mut self) -> Self {
+        self.ack_policy = Some(AckPolicy::None);
+        self
+    }
+
+    /// When acking a sequence number, this implicitly acks all sequences
+    /// below this one as well.
+    pub fn ack_all(mut self) -> Self {
+        self.ack_policy = Some(AckPolicy::All);
+        self
+    }
+
+    /// Requires ack or nack for all messages.
+    pub fn ack_explicit(mut self) -> Self {
+        self.ack_policy = Some(AckPolicy::Explicit);
+        self
+    }
+
+    /// Sets the number of redeliveries for a message.
+    pub fn max_deliver(mut self, n: i64) -> Self {
+        self.max_deliver = Some(n);
+        self
+    }
+
+    /// Sets the number of outstanding acks that are allowed before
+    /// message delivery is halted.
+    pub fn max_ack_pending(mut self, n: i64) -> Self {
+        self.max_ack_pending = Some(n);
+        self
+    }
+
+    /// Replays the messages at the original speed.
+    pub fn replay_original(mut self) -> Self {
+        self.replay_policy = Some(ReplayPolicy::Original);
+        self
+    }
+
+    /// Replays the messages as fast as possible.
+    pub fn replay_instant(mut self) -> Self {
+        self.replay_policy = Some(ReplayPolicy::Instant);
+        self
+    }
+
+    /// The bits per second rate limit applied to the push consumer.
+    pub fn rate_limit(mut self, n: u64) -> Self {
+        self.rate_limit = Some(n);
+        self
+    }
+    /// Specifies the consumer deliver subject.
+    ///
+    /// This option is used only in situations where the consumer does not exist
+    /// and a creation request is sent to the server.
+    ///
+    /// If not provided, an inbox will be selected.
+    pub fn deliver_subject(mut self, subject: String) -> Self {
+        self.deliver_subject = Some(subject);
+        self
+    }
+
+    /// Instruct the consumer to only deliver headers and no payloads.
+    pub fn headers_only(mut self) -> Self {
+        self.headers_only = Some(true);
+        self
+    }
+
+    /// Enables flow control
+    pub fn enable_flow_control(mut self) -> Self {
+        self.flow_control = Some(true);
+        self
+    }
+
+    /// Enables hearbeat messages to be sent.
+    #[allow(clippy::cast_possible_truncation)]
+    pub fn idle_heartbeat(mut self, interval: Duration) -> Self {
+        self.idle_heartbeat = Some(interval.as_nanos() as i64);
```

Comment:
Any reason to not use `Duration` as a type for `idle_hearbeat` ?

Reply:
Mainly just because the other fields are represented as integers.
So all of them have to be done for consistency and this is already quite a large change.

Reply:
I'll sort out the serialisation issues in a single commit when re-basing.

Reply:
They could go over wire as i64 if that is what other side needs but feels like a higher cognitive load for user converting between nanosecs and things that make sense to them.